### DISCUSSION TITLE
Custom build: Blank gray login screen after Nov 2 update

### DISCUSSION THREAD

**Discussion by aniqueta** [photoprism/photoprism#2847] [Q&A] [answered]
Hello, I've been following https://github.com/photoprism/photoprism/issues/98 and I'm super excited about the progress. 

In updating to the November 2 release, I've run into an issue where the login page ends up being just a blank gray page. 

I've gone through the steps of `photoprism users reset` and the log with `--debug` shows nothing out of the ordinary. Additionally, WebDAV import is working.

Looking at the page source, I see the following as the last lines of the page at `/library/login.` It looks like the rest of the page is not coming through. 

```
<div id="photoprism" class="splash-screen container theme-default">
  <noscript>
    <div id="noscript-info" class="splash-info">
      <a href="https://enable-javascript.com/" target="_blank">Please enable JavaScript in your browser.</a>
    </div>
  </noscript>
  <div id="splash-info" class="has-js splash-info">
```

I've tried multiple browsers, all with JS enabled and with extensions disabled. All produce the same. 

Would anyone have any suggestions on surfacing the issue here? Thank you!

**Comment by lastzero** [comment]
Using any proxy? Note the UI routes have changed.

**Reply by lastzero** [reply]
Note that the template names and variable names within those templates have changed. So if you are using custom templates or a custom assets path (which may be set automatically by Portainer), this could be an explanation.

**Comment by chx0003** [comment]
Exact same issue here after Nov 3 update.
Not using any proxy

**Reply by lastzero** [reply]
https://docs.photoprism.app/getting-started/troubleshooting/browsers/

Would you be able to debug this a little, for example check the logs?

**Reply by chx0003** [reply]
The log doesn't show any errors:
```
photoprism.service - PhotoPrism service
     Loaded: loaded (/etc/systemd/system/photoprism.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2022-11-03 14:50:27 UTC; 7s ago
    Process: 1126 ExecStart=/opt/photoprism/bin/photoprism up -d (code=exited, status=0/SUCCESS)
   Main PID: 1135 (photoprism)
      Tasks: 8 (limit: 38341)
     Memory: 78.3M
        CPU: 12.475s
     CGroup: /system.slice/photoprism.service
             -1135 /opt/photoprism/bin/photoprism up -d

Nov 03 14:50:20 TEST-SYS systemd[1]: Starting PhotoPrism service...
Nov 03 14:50:27 TEST-SYS photoprism[1126]: time="2022-11-03T14:50:27Z" level=info msg="PhotoPrism<C2><AE> needs your support!"
Nov 03 14:50:27 TEST-SYS photoprism[1126]: time="2022-11-03T14:50:27Z" level=info msg="Visit https://photoprism.app/membership to learn more."
Nov 03 14:50:27 TEST-SYS photoprism[1126]: time="2022-11-03T14:50:27Z" level=info msg="migrate: executing pre migrations"
Nov 03 14:50:27 TEST-SYS photoprism[1126]: time="2022-11-03T14:50:27Z" level=info msg="migrate: executing main migrations"
Nov 03 14:50:27 TEST-SYS photoprism[1126]: time="2022-11-03T14:50:27Z" level=info msg="closed database connection"
Nov 03 14:50:27 TEST-SYS photoprism[1126]: time="2022-11-03T14:50:27Z" level=info msg="daemon started with process id 1135\n"
Nov 03 14:50:27 TEST-SYS systemd[1]: Started PhotoPrism service.
```

And here's the body of the loaded frontpage:

```
<body class="settings nojs">

<div id="photoprism" class="splash-screen container theme-grayscale">
  <noscript>
    <div id="noscript-info" class="splash-info">
      <a href="https://enable-javascript.com/" target="_blank">Please enable JavaScript in your browser.</a>
    </div>
  </noscript>
  <div id="splash-info" class="has-js splash-info">
    </div></div></body>
```

Please let me know what else can I check.

**Reply by lastzero** [reply]
It seems you don't use our Docker image? In that case, you have to manually update / deploy the assets folder based on our image / install tar file used within that image.

**Reply by lastzero** [reply]
Kindly note we cannot provide free community support for custom builds.

**Reply by chx0003** [reply]
That's correct, I don't use the Docker image, I've compiled the files from the source code, but it was working before the update to Nov 3 release. I was using Sept 1 release without problems. After compiling and updating to the new version I updated the assets from the source files to the same path they were before (/opt/photoprism/assets). Also that path is defined as an environment variable PHOTOPRISM_ASSETS_PATH.
Is anything else that I should do after the upgrade?

Thanks and regards

**Reply by chx0003** [reply]
> Kindly note we cannot provide free community support for custom builds.

I understand, but I'm not using a custom build, just compiled everything from the source files in the release.

**Reply by lastzero** [reply]
Use the install target and install the tar file. Just compiling a new binary won't do.

**Reply by lastzero** [reply]
Do it exactly as we do it in the Dockerfiles provided as that is the only tested method.

**Reply by lastzero** [reply]
And make sure you use the same build tools and dependencies as those tiny little differences are what makes your build custom. It's not all in the source we provide.

**Comment by Donkeykong307** [comment, marked as answer]
I had the same issue. To fix it, I had to delete the old assets folder and copy the one that was made when I compiled the app.

**Comment by aniqueta** [comment]
Thanks. Using the new assets folder worked for me too.
